### PR TITLE
Support ARM64 platform

### PR DESCRIPTION
**Pull request by pakmarkthub**
Issue:
- See #63.

This PR:
- introduces GDRAPI_ARM64 definition.
- removes `tsc_khz` and `tm_cycles` when compiling on ARM64.
- marks (just a comment) `tm_cycles` and `cycles_per_ms` as deprecated in gdrdrv.h.

Presubmit testing:
- On gc11 (new gdrdrv with libgdrapi v2.1 and vice versa).
- On hpe-arm-07-01.

### CODE REVIEW

**Review by drossetti — CHANGES_REQUESTED** [review]

**Comment by drossetti** [comment]
README.md should include a reference to Arm64 support where appropriate, for example:

> Developed and tested on RH7.x and Ubuntu18_04. The supported architectures are Linux x86_64 and ppc64le.

File: src/gdrapi.c [hunk 1]
```diff
@@ -291,8 +291,10 @@ int gdr_get_info(gdr_t g, gdr_mh_t handle, gdr_info_t *info)
         info->va          = params.va;
         info->mapped_size = params.mapped_size;
         info->page_size   = params.page_size;
+        #ifndef GDRAPI_ARM64
```
[the remark this answers is not in the document]
Comment (pakmarkthub) [reply src/gdrapi.c]: As those two fields are available on all supported platforms, I just simply read from gdrdrv.

File: src/gdrdrv/gdrdrv.c [hunk 2]
```diff
@@ -33,10 +33,13 @@
 #include <linux/list.h>
 #include <linux/mm.h>
 #include <linux/io.h>
-#include <linux/timex.h>
-#include <linux/timer.h>
 #include <linux/sched.h>
 
+#ifndef CONFIG_ARM64
+    #include <linux/timex.h>
```
[the remark this answers is not in the document]
Comment (pakmarkthub) [reply src/gdrdrv/gdrdrv.c]: Fixed.

File: src/gdrdrv/gdrdrv.c [hunk 3]
```diff
@@ -238,8 +254,10 @@ struct gdr_mr {
     enum { GDR_MR_NONE, GDR_MR_WC, GDR_MR_CACHING } cpu_mapping_type;
     nvidia_p2p_page_table_t *page_table;
     int cb_flag;
+    #ifndef CONFIG_ARM64
```
[the remark this answers is not in the document]
Comment (pakmarkthub) [reply src/gdrdrv/gdrdrv.c]: Fixed.

File: src/gdrdrv/gdrdrv.c [hunk 4]
```diff
@@ -744,8 +771,10 @@ static int gdrdrv_get_info(gdr_info_t *info, void __user *_params)
     params.va          = mr->va;
     params.mapped_size = mr->mapped_size;
     params.page_size   = mr->page_size;
+    #ifndef CONFIG_ARM64
     params.tm_cycles   = mr->tm_cycles;
     params.tsc_khz     = mr->tsc_khz;
+    #endif
```
[the remark this answers is not in the document]
Comment (pakmarkthub) [reply src/gdrdrv/gdrdrv.c]: Fixed. We have `memset(0)` when creating mr.

File: src/gdrdrv/gdrdrv.h [hunk 5]
```diff
@@ -86,8 +86,10 @@ struct GDRDRV_IOC_GET_INFO_PARAMS
     __u64 va;
     __u64 mapped_size;
     __u32 page_size;
+    #if !(defined(CONFIG_ARM64) || defined(GDRAPI_ARM64))
```
[the remark this answers is not in the document]
Comment (pakmarkthub) [reply src/gdrdrv/gdrdrv.h]: Fixed.

File: src/gdrdrv/gdrdrv.c [hunk 6]
```diff
@@ -124,8 +127,21 @@ static inline int gdr_pfn_is_ram(unsigned long pfn)
 #endif
 }
 
+#elif defined(CONFIG_ARM64)
+static inline pgprot_t pgprot_modify_writecombine(pgprot_t old_prot)
+{
+    return pgprot_writecombine(old_prot);
```
[the remark this answers is not in the document]
Comment (pakmarkthub) [reply src/gdrdrv/gdrdrv.c:130]: Confirmed on linux kernel version 3 - 5. `pgprot_writecombine` is a macro. That macro uses only non-GPL symbols.

**Comment by pakmarkthub** [comment]
> README.md should include a reference to Arm64 support where appropriate, for example:
> 
> > Developed and tested on RH7.x and Ubuntu18_04. The supported architectures are Linux x86_64 and ppc64le.

Updated.

@drossetti Can I ask you to re-review?

**Review by drossetti — APPROVED** [review]

**Pull request merged**